June 5, 2024 - add Vandit Singh contributor spotlight (#138)
jenkins-infra/contributor-spotlight: src/pages/contributors/vandit-singh.adoc

= Vandit Singh
:page-name: Vandit Singh
:page-linkedin: vandit-singh
:page-twitter: vandittweets
:page-github: Vandit1604
:page-email: 
:page-image: avatar/vandit-singh.png
:page-pronouns: He/Him/His
:page-location: Delhi NCR, India
:page-firstcommit: 2022
:page-datepublished: 2024-06-05
:page-featured: true

> :page-intro:  Vandit Singh is a Jenkins contributor currently based in India. During the day, he enjoys tinkering with code. When he's not deep in Jenkins code, Vandit loves creating music or strumming his guitar. Open source and music are very close to his heart. Vandit's tech journey started when he learned Java and started exploring open source by contributing to Jenkins. He had developed a particular interest in DevOps during his first year of university, and Jenkins seemed like the perfect starting point. At first, the large codebase scared him. However, by taking one step at a time, Vandit keeps progressing.

Reviewer: ```suggestion
:page-intro:  Vandit Singh is a Jenkins contributor currently based in India. During the day, he enjoys tinkering with code. When he's not deep in Jenkins code, Vandit loves to create music or strum his guitar. Open source and music are very close to his heart. Vandit's tech journey started when he learned Java and started to explore open source by contributing to Jenkins. He had developed a particular interest in DevOps during his first year of university, and Jenkins seemed like the perfect starting point. At first, the large codebase scared him. However, by taking one step at a time, Vandit keeps progressing.
```

> :page-intro:  Vandit Singh is a Jenkins contributor currently based in India. During the day, he enjoys tinkering with code. When he's not deep in Jenkins code, Vandit loves creating music or strumming his guitar. Open source and music are very close to his heart. Vandit's tech journey started when he learned Java and started exploring open source by contributing to Jenkins. He had developed a particular interest in DevOps during his first year of university, and Jenkins seemed like the perfect starting point. At first, the large codebase scared him. However, by taking one step at a time, Vandit keeps progressing.

Reviewer 2: ```suggestion
:page-intro:  Vandit Singh is a Jenkins contributor currently based in India.
During the day, he enjoys tinkering with code.
When he's not deep in Jenkins code, Vandit loves creating music or strumming his guitar.
Open source and music are very close to his heart.
Vandit's tech journey started when he learned Java and started exploring open source by contributing to Jenkins.
He had developed a particular interest in DevOps during his first year of university, and Jenkins seemed like the perfect starting point.
At first, the large codebase scared him.
However, by taking one step at a time, Vandit keeps progressing.
```

== What is your background prior to contributing to Jenkins?

Before contributing to Jenkins, I had little knowledge of CI/CD, DevOps, and Java.

> While in my Freshman year of university, with only basic tech knowledge, it was during this time that the love for open source was instilled into me by my seniors and university mates.

Reviewer: ```suggestion
During my Freshman year at university, with only basic tech knowledge, the love for open source was instilled into me by my seniors and university mates.
```

> While in my Freshman year of university, with only basic tech knowledge,  it was during this time that the love for open source was instilled into me by my seniors and university mates.

Reviewer 2: ```suggestion
While in my Freshman year of university, with only basic tech knowledge, it was during this time that the love for open source was instilled into me by my seniors and university mates.
```

I have been contributing to Jenkins since July 2022.

== Why choose Jenkins over other projects?

One of the most exciting aspects for me is the idea of creating something that can be used by other people. 
The bar is not very high for anyone to start contributing.
If you can code, you can get involved.
If you can design, you're welcome in the community.
You can write?
Awesome!
You help us with the documentation.
The ease of getting involved with the community was a deciding factor for me in choosing to work with Jenkins over other projects.

== What problems has Jenkins solved for you?

Jenkins is used for CI/CD pipelining and automating tasks on the internet.
You can automate many tasks in your web app with Jenkins.
Additionally, Jenkins is used to improve the software development cycle by testing new pull requests every day.

== Is there an aspect of Jenkins that you're particularly passionate about?

I'm passionate about Jenkins Infrastructure and the Remoting component of Jenkins.

== What sort of contributions have felt the most successful or impactful?

The work I did in Google of Summer of Code 2023 seems the most successful for me.

> I got to link:https://www.jenkins.io/projects/gsoc/2023/projects/alternative-jenkinsio-build-tool/[build a complete project from scratch] that will be used by many Jenkins users.

Reviewer 3: ```suggestion
I got to link:https://www.jenkins.io/projects/gsoc/2023/projects/alternative-jenkinsio-build-tool/[adopt a different tooling for the jenkins.io website] that will be used by many Jenkins users.
```

Reviewer 3: I don't think the original wording is accurate, since we only started changing the tooling, but not the contents of the jenkins.io website during GSoC 2023.

> I got to link:https://www.jenkins.io/projects/gsoc/2023/projects/alternative-jenkinsio-build-tool/[build a complete project from scratch] which will be used by many users of Jenkins.

Reviewer: ```suggestion
I got to link:https://www.jenkins.io/projects/gsoc/2023/projects/alternative-jenkinsio-build-tool/[build a complete project from scratch] that will be used by many Jenkins users.
```

== Advice for new developers and new members of the open-source community

> The vastness of Jenkins' codebase is a boon for all but can confuse new contributors. 

Reviewer: ```suggestion
The vastness of Jenkins' codebase is a boon for all, but can confuse new contributors. 
```
